lms/djangoapps/courseware/courses.py

```diff
@@ -109,12 +112,13 @@ def get_course_overview_with_access(user, action, course_key, check_if_enrolled=
     return course_overview
 
 
-def check_course_access(course, user, action, check_if_enrolled=False):
+def check_course_access(course, user, action, check_if_enrolled=False, check_survey_complete=True):
     """
     Check that the user has the access to perform the specified action
     on the course (CourseDescriptor|CourseOverview).
 
     check_if_enrolled: If true, additionally verifies that the user is enrolled.
+    check_survey_complete: If true, additionally verifies that the user has completed the survey.
     """
```

Comment:
FYI: @nasthagiri: We need to land this story and although this parameter is ugly, it is cleaner than it was and one step closer to consolidating permissions.  It would be great to continue to chip away at this and have a proper design we know we want to move toward.